src/main/java/com/google/devtools/build/lib/worker/WorkRequestHandler.java

```diff
@@ -311,6 +315,8 @@ public WorkRequestHandler build() {
    * returns. If {@code in} reaches EOF, it also returns.
    */
   public void processRequests() throws IOException {
+    WorkerIO workerIO = WorkerIO.redirectSystemStreams();
```

Comment:
Looking at https://github.com/bazelbuild/bazel/pull/14623, you're not the only one with this idea. You might be right in making it optional, as some workers do their own handling. Fortunately that could easily be done by adding a method to the builder.